### ISSUE TITLE
rsyslog 8.2110.0 openssl implementation on AIX

### ISSUE THREAD

**Issue by LFrank2021** [rsyslog/rsyslog#4799]
I am currently unable to send messages using openssl StreamDriver.
The compilation runs without error, though a lot of warnings.
I am trying to use the IBM distributed openSSL Implementation.
Without the exported OPENSSL_CFLAGS and OPENSSL_LIBS the pkgconf give an error:

> configure: error: Package requirements (openssl) were not met:
> 
> No package 'openssl' found
> 
> Consider adjusting the PKG_CONFIG_PATH environment variable if you
> installed software in a non-standard prefix.
> 
> Alternatively, you may set the environment variables OPENSSL_CFLAGS
> and OPENSSL_LIBS to avoid the need to call pkg-config.
> See the pkg-config man page for more details.

### Expected behavior
sending messages through StreamDriver="ossl" StreamDriverMode="1" StreamDriverAuthMode="anon"

### Actual behavior
Errormessage:

> rsyslogd: could not load module 'lmnsd_ossl', 
> errors: trying to load module /opt/freeware/lib/rsyslog/lmnsd_ossl.so: 
> rtld: 0712-001 Symbol CRYPTO_num_locks was referenced       from module /opt/freeware/lib/rsyslog/lmnsd_ossl.so(), but a runtime definition      of the symbol was not found. 
> rtld: 0712-001 Symbol CRYPTO_set_id_callback was referenced       from module /opt/freeware/lib/rsyslog/lmnsd_ossl.so(), but a runtime definition      of the symbol was not found. 
> rtld: 0712-001 Symbol CRYPTO_set_locking_callback was referenced       from module /opt/freeware/lib/rsyslog/lmnsd_ossl.so(), but a runtime definition      of the symbol was not found. 
> rtld: 0712-001 Symbol CRYPTO_set_dynlock_create_callback was referenced       from module /opt/freeware/lib/rsyslog/lmnsd_ossl.so(), but a runtime definition      of the symbol was not found. 
> rtld: 0712-001 Symbol CRYPTO_set_dynlock_lock_callback was referenced       from module /opt/freeware/lib/rsyslog/lmnsd_ossl.so(), but a runtime definition      of the symbol was not found. 
> rtld: 0712-001 Symbol CRYPTO_set_dynlock_destroy_callback was referenced       from module /opt/freeware/lib/rsyslog/lmnsd_ossl.so(), but a runtime definition      of the symbol was not found. 
> rtld: 0712-001 Symbol ERR_get_error was referenced       from module /opt/freeware/lib/rsyslog/lmnsd_ossl.so(), but a runtime definition      of the symbol was not found.

### Steps to reproduce the behavior
compile under AIX 7200-02-05 using xlC 16.1 using xlclang++ compiler

### Environment
- rsyslog version: 8.2110.0
- platform: AIX 7.2 SP2 TL5
- compiler: xlclang++ (16.1)
- compile evironment:
> export CONFIG_SHELL=/opt/freeware/bin/bash
> export CONFIG_ENV_ARGS=/opt/freeware/bin/bash
> 
> export CC=xlclang++
> export CFLAGS="-qmaxmem=16384 -DSYSV -D_AIX -D_AIX32 -D_AIX41 -D_AIX43 -D_AIX51 -D_AIX52 -D_AIX53 -D_AIX61 -D_AIX71 -D_AIX72 -D_ALL_SOURCE -DFUNCPROTO=15 -O2 -I/opt/freeware/include -I/usr/include"
> 
> export CXX=xlclang++
> export CXXFLAGS=$CFLAGS
> 
> export F77=xlf
> export FFLAGS="-qmaxmem=16384 -O -I/opt/freeware/include"
> 
> export LD=ld
> export LDFLAGS="-L/opt/freeware/lib -Wl,-blibpath:/opt/freeware/lib:/usr/lib:/lib -Wl,-bmaxdata:0x80000000"
> export LDFLAGS="-L/opt/freeware/lib64 -L/opt/freeware/lib -Wl,-blibpath:/opt/freeware/lib64:/opt/freeware/lib:/usr/lib:/lib -Wl,-bmaxdata:0x80000000"
> 
> export PATH=/opt/freeware/bin:/opt/freeware/sbin:/usr/bin:/bin:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/vac/bin:/usr/vacpp/bin:/usr/ccs/bin:/usr/dt/bin:/usr/opt/perl5/bin:/usr/local/bin:/usr/lib/instl
> export PATH=/opt/IBM/xlC/16.1.0/bin:$PATH
> 
> export OPENSSL_CFLAGS="$CFLAGS"
> export OPENSSL_LIBS="/usr/lib/"

- for configuration questions/issues, include rsyslog.conf and included config files
> *.*                             action(type="omfwd" Target="MyRsyslogServer" Port="MyPort" Protocol="tcp" Template="RSYSLOG_SyslogProtocol23Format" StreamDriver="ossl" StreamDriverMode="1" StreamDriverAuthMode="anon")

**Comment by LFrank2021** [comment]
I think I found a way to make it work:
AIX by default seems to use /usr/lib/pkgconfig for .pc files
When using the OpenSource pkg-config rpm it uses /opt/freeware/lib/pkgconfig/
For now I copied the .pc files for openssl, libcrypto and libssl to /opt/freeware/lib/pkgconfig/ and the compilation works and the error message is no longer thrown.
Strangely today even the PKG_CONFIG_PATH works....
Well, no bug but a layer 8 problem.
But maybe this is helpful for someone else as well.

Thank you
